A29-xds-tls-security.md

```diff
@@ -0,0 +1,562 @@
+[RFC6125-6]: https://datatracker.ietf.org/doc/html/rfc6125#section-6
+[zero-compr]: https://datatracker.ietf.org/doc/html/rfc5952#section-2.2
+[no-leading-0]: https://datatracker.ietf.org/doc/html/rfc5952#section-2.1
+[lower-case]: https://datatracker.ietf.org/doc/html/rfc5952#section-2.3
+[DNS-wildcard]: https://github.com/envoyproxy/envoy/blob/c94e646e0280e4c521f8e613f1ae2a02b274dbbf/api/envoy/extensions/transport_sockets/tls/v3/common.proto#L358-L360
+
+#### LDS for Server Side Security
+
+Server-side xDS processing is described in [A36: xDS-Enabled Servers][A36].
+A Listener resource has zero or more [`filter_chains`][filter-chains] and one of those
+`filter_chain`s is matched against an incoming connection (or the
+[`default_filter_chain`][default-filter-chain] if there is no match) in order to get and apply
+the security (and other) configuration for that connection as described in the
+[`FilterChainMatch`][filter-chain-match] section. The [`transport_socket`][transport-socket]
+of the matched (selected) `filter_chain` is used to extract the
+[`DownstreamTlsContext`][DTC] as described [here][transport-socket-comment].
+As part of validating all LDS updates, if a `transport_socket` name is not
+`envoy.transport_sockets.tls` i.e. something we don't recognize, gRPC will
+NACK an LDS update. Otherwise the `DownstreamTlsContext` thus obtained is used
+for the incoming connection if the server is using `XdsServerCredentials`.
+If there is no `DownstreamTlsContext` (such as when [`transport_socket`][transport-socket]
+is not present) then gRPC uses the fallback credentials for the incoming connection.
+
+[`common_tls_context`][CTC1] in the `DownstreamTlsContext` contains the required
+security configuration. See below for [`CommonTlsContext`][CTC-type] processing.
+
+A secure server at a minimum requires an identity certificate which is provided
+by [`tls_certificate_certificate_provider_instance`][TCCPI]. If that field is
+empty, gRPC will NACK an update as part of validating all LDS updates. If
+[`validation_context_certificate_provider_instance`][VCCPI1] inside
+[`combined_validation_context`][CVC] is provided then the server will request
+the client certificate. When [`require_client_certificate`][RCC] is `true`,
+gRPC requires the client certificate and will reject a connection without a valid
+client certificate. When [`require_client_certificate`][RCC] is `true`, the
+[`validation_context_certificate_provider_instance`][VCCPI1] value should be
+present otherwise gRPC will NACK an LDS update as part of validating all LDS
+updates.
+
+The following fields in an LDS update are ignored:
+
+* [disable_stateless_session_resumption][DIS-STATELESS-SESS-RES]
+* [session_timeout][SESSION-TIMEOUT]
+
+If any of the following fields are present, gRPC will NACK an LDS update as
+part of validating all LDS updates:
+
+* [require_sni][REQ-SNI]
+* [session_ticket_keys][SESSION-TICKET-KEYS]
```

Comment:
as discussed, it seems fine to `ignore session_ticket_keys` and `session_ticket_keys_sds_secret_config`

Reply:
yes, changed